botocore/__init__.py

```diff
@@ -70,6 +67,11 @@ def xform_name(name, sep='_', _xform_cache=_xform_cache):
         return name
     key = (name, sep)
     if key not in _xform_cache:
+        if _special_case_transform.search(name) is not None:
+            is_special = _special_case_transform.search(name)
```

Comment:
How about switch these two lines like this? It seems cleaner, and probably slightly faster.

    is_special = _special_case_transform.search(name)
    if is_special:
        blah_blah